libs/common/src/change-stream/test-helpers/add-audit-record-to-expected-response.ts

```diff
@@ -0,0 +1,23 @@
+import { AuditDatabaseRecord, AuditDetails } from '../../types';
+import { generateParsedMockAuditDatabaseRecord } from './generate-mock-audit-database-record';
+
+type AddAuditRecordToResponseParams =
+  | {
+      baseResponse: object;
+      auditRecord: AuditDatabaseRecord;
+    }
+  | {
+      baseResponse: object;
+      auditDetails: AuditDetails;
+    };
+
+export const addAuditRecordToExpectedResponse = ({ baseResponse, ...auditObject }: AddAuditRecordToResponseParams) => {
+  if ('auditRecord' in auditObject) {
+    return {
+      ...baseResponse,
+      auditRecord: auditObject.auditRecord,
+    };
+  }
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  return { ...baseResponse, auditRecord: generateParsedMockAuditDatabaseRecord(auditObject.auditDetails) };
+};
```

Comment:
FWIW could do the following, but happy to just ship this
```suggestion
type AddAuditRecordToResponseParams =
  | {
      baseResponse: object;
      auditRecord: AuditDatabaseRecord;
      auditDetails?: undefined;
    }
  | {
      baseResponse: object;
      auditRecord?: undefined;
      auditDetails: AuditDetails;
    };

export const addAuditRecordToExpectedResponse = ({ baseResponse, ...auditObject }: AddAuditRecordToResponseParams) => {
  if (auditObject.auditRecord) {
    return {
      ...baseResponse,
      auditRecord: auditObject.auditRecord,
    };
  }
  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
  return { ...baseResponse, auditRecord: generateParsedMockAuditDatabaseRecord(auditObject.auditDetails) };
};
```

Reply:
Or alternatively, if happy decoupling the existence of AuditRecord/Details
```
type AddAuditRecordToResponseParams =
  | {
      baseResponse: object;
      auditRecord: AuditDatabaseRecord;
      auditDetails?: undefined;
    }
  | {
      baseResponse: object;
      auditRecord?: undefined;
      auditDetails: AuditDetails;
    };

export const addAuditRecordToExpectedResponse = ({ baseResponse, auditDetails, auditRecord}: AddAuditRecordToResponseParams) => {
  if (auditRecord) {
    return {
      ...baseResponse,
      auditRecord: auditRecord,
    };
  }
  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
  return { ...baseResponse, auditRecord: generateParsedMockAuditDatabaseRecord(auditDetails) };
};

```